src/rust/engine/client/src/pantsd.rs

```diff
@@ -0,0 +1,257 @@
+// Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+// Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+use std::path::{Path, PathBuf};
+use std::{fmt, fs};
+
+use libc::pid_t;
+use log::debug;
+use sha2::{Digest, Sha256};
+
+struct Metadata {
+  metadata_dir: PathBuf,
+}
+
+impl Metadata {
+  fn mount<P: AsRef<Path>>(directory: P) -> Result<Metadata, String> {
+    let info = uname::uname().map_err(|e| format!("{}", e))?;
+    let host_hash = Sha256::new()
+      .chain(&info.sysname)
+      .chain(&info.nodename)
+      .chain(&info.release)
+      .chain(&info.version)
+      .chain(&info.machine)
+      .finalize();
+
+    const HOST_FINGERPRINT_LENGTH: usize = 12;
+    let mut hex_digest = String::with_capacity(HOST_FINGERPRINT_LENGTH);
+    for byte in host_hash {
+      fmt::Write::write_fmt(&mut hex_digest, format_args!("{:02x}", byte)).unwrap();
+      if hex_digest.len() >= HOST_FINGERPRINT_LENGTH {
+        break;
+      }
+    }
+
+    let metadata_dir = directory
+      .as_ref()
+      .join(&hex_digest[..HOST_FINGERPRINT_LENGTH])
+      .join("pantsd");
+    if metadata_dir.is_dir() {
+      Ok(Metadata { metadata_dir })
+    } else {
+      Err(format!(
+        "There is no pantsd metadata at {metadata_dir}.",
+        metadata_dir = metadata_dir.display()
+      ))
+    }
+  }
+
+  fn pid(&self) -> Result<pid_t, String> {
```

Comment:
Nit: `remoteprocess` has a typedef for this that would avoid (explicitly) depending on libc: https://docs.rs/remoteprocess/0.4.3/remoteprocess/type.Pid.html

Reply:
Yeah, as does nix. The pantsd module does not depend on any of this though so I'll switch to the not so intuitive i32 which they're all a typedef for.